File: introduction/ITP1_6_D-Matrix Vector Multiplication.c
#include <stdio.h>

int main() {
  int n, m, i, j, sum;
  int a[100][100];
  int b[100];
  scanf("%d %d", &n, &m);
  for(i = 0; i < n; i++) for(j = 0; j < m; j++) scanf("%d", &a[i][j]);
  for(j = 0; j < m ; j++) scanf("%d",&b[j]);
  for(i = 0; i < n; i++) {
    sum = 0;
    for(j = 0; j < m; j++) sum += a[i][j] * b[j];
    printf("%d\n", sum);
  }
  return 0;
}

File: introduction/ITP1_6_D-Matrix Vector Multiplication.cs
using System;
using System.Collections.Generic;

public class Program {
  public static void Main() {
    var line = Console.ReadLine().Split();
    int n = int.Parse(line[0]), m = int.Parse(line[1]);
    var arr = new List<string>();
    while((line[0] = Console.ReadLine()) != null) arr.Add(line[0]);
    for(int i = 0; i < n; i++) {
      int sum = 0;
      for(int j = 0; j < m; j++) sum += int.Parse(arr[i].Split()[j]) * int.Parse(arr[n+j]);
      Console.WriteLine(sum);
    }
  }
}
